src/controller/CHIPDeviceController.cpp

```diff
@@ -2062,7 +2051,16 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
     }
     break;
     case CommissioningStage::kFindOperational: {
-        CHIP_ERROR err = UpdateDevice(proxy->GetDeviceId());
+        const PeerId peerId              = mFabricInfo->GetPeerIdForNode(proxy->GetDeviceId());
+        OperationalDeviceProxy * session = GetDeviceSession(peerId);
+        if (session == nullptr)
+        {
+            ChipLogError(Controller, "No active session to find an operational address for.");
+            CommissioningStageComplete(CHIP_ERROR_NOT_FOUND);
+            return;
+        }
+
+        CHIP_ERROR err = session->LookupPeerAddress();
```

Comment:
```suggestion
        CHIP_ERROR err = UpdateDevice(proxy->GetDeviceId());
```
is the same thing, no?